### PR TITLE
Add GEDCOM genealogy data file support

### PR DESCRIPTION
**Pull request by fguitton**
I am opening this PR to bring Genealogy Data File (GEDCOM) to linguist. The format has been around since 1984 and has received official Release in 2019. Background information can be found there:
  - https://en.wikipedia.org/wiki/GEDCOM
  - https://www.gedcom.org/gedcom.html

## Checklist:
- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension (~65K):
      - https://github.com/search?q=extension%3Aged+NOT+nothack&type=Code
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://edge.fscdn.org/assets/img/documents/ged551-5bac5e57fe88dd37df0e153d9c515335.pdf
      - https://famousfamilytrees.blogspot.com/2008/07/fictional-family-trees.html
    - Sample license(s):
      - The first sample is imported from the Official 5.5.1 Specification document open for public consultation
      - The second sample is offered by Fictional Family Trees and provided freely
  
  - [x] I have included a syntax highlighting grammar: [https://github-lightshow.herokuapp.com/](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fgithub.com%2Ffguitton%2Fvscode-gedcom%2Fblob%2Fmaster%2Fsyntaxes%2Fgedcom.tmLanguage.json&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Ffguitton%2Flinguist%2Fblob%2Flang%2Fgedcom%2Fsamples%2FGEDCOM%2FHarryPotter.ged&code=)

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -1619,6 +1619,14 @@ GDScript:
   tm_scope: source.gdscript
   ace_mode: text
   language_id: 123
+GEDCOM:
+  type: data
+  ace_mode: text
+  extensions:
+  - ".ged"
+  - ".gedcom"
```
Comment (pchaigno) [inline lib/linguist/languages.yml:1627]: That second extension only has 10 files on GitHub and therefore doesn't match our criteria of *hundreds of repositories*.

**Comment by fguitton** [comment]
> What makes you think they are okay to use?

Thank you for reviewing @pchaigno !

I've gone ahead and switched the sample files to a couple available directly on the GEDCOM website (https://www.gedcom.org/samples.html). I could not find license notices for them, but I hope that it is satisfactory. If not I could replace them with the example available on the Wikipedia entry for GEDCOM (https://en.wikipedia.org/wiki/GEDCOM) or one that I could generate myself whichever would be your preference.

**Comment by pchaigno** [comment]
> I've gone ahead and switched the sample files to a couple available directly on the GEDCOM website (https://www.gedcom.org/samples.html). I could not find license notices for them, but I hope that it is satisfactory. If not I could replace them with the example available on the Wikipedia entry for GEDCOM (https://en.wikipedia.org/wiki/GEDCOM) or one that I could generate myself whichever would be your preference.

What about selecting a couple from GitHub.com after checking the license of the repositories they live in?

**Comment by fguitton** [comment]
> What about selecting a couple from GitHub.com after checking the license of the repositories they live in?

I have now added the file [Royal92.ged](https://github.com/altesack/gencms/blob/master/gedcomfiles/Royal92.ged) from the MIT LICENSED repo https://github.com/altesack/gencms. The 1992 note from the original author Denis Reid, included in the file, mentions he is happy for the document to be made "[available to whomever](https://github.com/altesack/gencms/blob/master/gedcomfiles/Royal92.ged#L20)"

**Pull request merged**